src/app/trainees/trainee-search/trainee-search.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from "@angular/core";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { ActivatedRoute, Params } from "@angular/router";
+import { Select, Store } from "@ngxs/store";
+import { Observable } from "rxjs";
+import { take } from "rxjs/operators";
+import {
+  GetTrainees,
+  SearchTrainees,
+  UpdateTraineesRoute
+} from "../state/trainees.actions";
+import { TraineesState } from "../state/trainees.state";
+
+@Component({
+  selector: "app-trainee-search",
+  templateUrl: "./trainee-search.component.html"
+})
+export class TraineeSearchComponent implements OnInit {
+  @Select(TraineesState.searchQuery) searchQuery$: Observable<string>;
+  public form: FormGroup;
+  public params: Params = this.route.snapshot.queryParams;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private store: Store,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.setupForm();
+  }
+
+  public setupForm(): void {
+    this.form = this.formBuilder.group({
+      searchQuery: [this.params.searchQuery, [Validators.minLength(2)]]
```

Comment:
![Screenshot 2020-04-14 at 16 10 52](https://user-images.githubusercontent.com/13540819/79241400-eb726f80-7e6a-11ea-9c7b-cdb864df4943.png)

please confirm and I'll approve.

Reply:
Doubt this meant validation but rather Observable debounce and check value length >= 2.

I may be off, but confirm and I'll approve.

Reply:
ah good spot, will clarify with Ify